docs/2.8.0/docs/daml/reference/templates.rst

```diff
@@ -92,7 +92,58 @@ Template-local Definitions
    - interface methods applied to ``this``
 
    in a template-local let definition will result in an error generated by the infinite loop.
- 
+
+.. warning::
+   Since Daml 2.8.0, template-local definitions are deprecated and their
+   presence will result in the following warning:
+
+   .. code-block:: text
+
+     Template-local binding syntax ("template-let") is deprecated,
+     it will be removed in a future version of Daml.
+     Instead, use plain top level definitions, taking parameters
+     for the contract fields or body ("this") if necessary.
+
+Migrating
+=========
+
+Users are strongly encouraged to rewrite their code to avoid this feature. Such
+a rewrite would involve replacing each template-local definition with a regular
+top level definition. If the old definition made use of contract fields or the
+contract body ("this"), those won't be in scope at the top level, so the new
+definition should take them as parameters. Correspondingly, the use sites of
+these definitions should supply the appropriate values as arguments.
+
+Disabling the warning
+=====================
+
+This warning is controlled by the warning flag ``template-let``, which means
+that it can be disabled independently of other warnings. This is especially
+useful for gradually migrating code that used this syntax.
+
+To disable the warning within a Daml file, add the following line at the top of
+the file:
+
+.. code-block:: daml
+
+  {-# OPTIONS_GHC -Wno-template-let #-}
+
+To disable it for an entire Daml project, add the following entry to the
+``build-options`` field of the project's ``daml.yaml`` file
+
+.. code-block:: yaml
+
+  build-options:
+  - --ghc-option=-Wno-template-let
+
+Within a project where the warning has been disabled as described above, it can
+be reenabled for a single Daml file by adding the following line at the top of
```

Comment:
```suggestion
Within a project where the warning has been deactivated for the whole project as described above, it can
be reactivated for a single Daml file by adding the following line at the top of
```